evaluator/diagnostic_converter/test/test_converter_node.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "converter_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "diagnostic_msgs/msg/diagnostic_array.hpp"
+#include "diagnostic_msgs/msg/diagnostic_status.hpp"
+#include "diagnostic_msgs/msg/key_value.hpp"
+#include <tier4_simulation_msgs/msg/user_defined_value.hpp>
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+using ConverterNode = diagnostic_converter::DiagnosticConverter;
+using diagnostic_msgs::msg::DiagnosticArray;
+using diagnostic_msgs::msg::DiagnosticStatus;
+using diagnostic_msgs::msg::KeyValue;
+using tier4_simulation_msgs::msg::UserDefinedValue;
+
+void waitForMsg(
+  bool & flag, const rclcpp::Node::SharedPtr node1, const rclcpp::Node::SharedPtr node2)
+{
+  while (!flag) {
+    rclcpp::spin_some(node1);
+    rclcpp::spin_some(node2);
+    rclcpp::sleep_for(std::chrono::milliseconds(100));
+  }
+  flag = false;
+}
+
+std::function<void(UserDefinedValue::ConstSharedPtr)> generateCallback(
+  bool & flag, UserDefinedValue & msg)
+{
+  return [&](UserDefinedValue::ConstSharedPtr received_msg) {
+    flag = true;
+    msg = *received_msg;
+  };
+}
+
+TEST(ConverterNode, ConvertDiagnostics)
+{
+  const std::vector<std::string> input_topics = {"/test1/diag", "/test2/diag"};
+
+  rclcpp::init(0, nullptr);
+  rclcpp::Node::SharedPtr dummy_node = std::make_shared<rclcpp::Node>("converter_test_node");
+
+  rclcpp::NodeOptions options;
+  options.append_parameter_override("diagnostic_topics", rclcpp::ParameterValue(input_topics));
```

Comment:
This test is broken because the `input_topics` which are set as parameter `diagnostic_topics` are not used by the node.